onnxruntime/core/providers/cuda/tensor/scatter_nd_impl.cu

```diff
@@ -34,9 +34,12 @@ __global__ void _ScatterNDKernel(
     // This would have been an error in the CPU kernel, but throwing in the CUDA EP
     // is hard. This is the approach taken by other frameworks for out of bound indices
     // in their corresponding GPU backends as well.
-    if (index < 0)
-      index = 0;
+    // index >= -dim_value && index < dim_value
 
+    if (index < -dim_value) 
```

Comment:
Same comment here, if `index >= 0`, 3 tests are done. If this test is moved to the first place, only 2 are needed in all cases.

Reply:
done.